### DISCUSSION TITLE
Development on Windows 10

### DISCUSSION THREAD

**Discussion by lucadalli** [google/model-viewer#2690] [Q&A]
I'm facing some difficulties getting the tooling to work on Windows 10.  I'm using WSL2 (Ubuntu 18.04.5) as advised but the tests fail to run and the dev command does not seem to build. I'm having to rebuild manually after every change :(

I've tried gone through #1121 and its linked issues, to no avail.

Here's a few questions which might help guide me in the right direction
1. Why does `model-viewer`'s `serve` command (which is part of `dev`) serve the local directory? It appears to me that nothing meaningful like examples with model-viewer instances can be reached that way.
Unlike the `serve` command in the root `package.json` which serves all packages including the docs, where I can find some `model-viewer` instances. Should I be using the latter `serve` command instead?

2. Is this output typical for the `model-viewer`'s `dev` command? After changing a file, the process seems to detect the change but does not output which file changed and nor does it seem to actually build.

![image](https://user-images.githubusercontent.com/39853447/129114794-55b6450a-8bae-425a-82cf-7c37bbc7173a.png)

**Comment by elalish** [comment]
Actually, I haven't ever bothered with `dev`, possibly because it wasn't working well for me either. I tend to keep a terminal open to the model-viewer sub package and just do `npm run build:dev` occasionally when I want to test something. Likewise I have a separate terminal running `npm run serve`, which I just leave alone. @timmmeh, have you ever used `dev`? We develop in VSCode, so its auto syntax checkers are usually catching the errors as I type without having to actually build constantly. Does any of that help?

**Reply by lucadalli** [reply]
Interesting. That's the same workflow I've been using so far.

I'm also using VSCode. I installed the `clang-format` extension but it does not add semi-colons nor recognize the script tags in .html files. Is this normal?

**Reply by elalish** [reply]
True, it doesn't add semi-colons and we normally have to turn off auto-format for HTML, since it messes up the examples by tabbing all the code over. If you happen to know easy fixes for these things, that would be great, but I've just gotten used to it by now... 😂

**Reply by lucadalli** [reply]
Ah, ok. I'm afraid I don't but they're only minor nuisances. As long as it's not just me, I can live with them 😁

**Comment by lucadalli** [comment]
For anyone facing a `No binary for ChromeHeadless browser on your platform. Please, set "CHROME_BIN" env variable` error when attempting to run tests:

The easiest fix I found was setting `CHROME_BIN` as the mounted Windows Chrome like so:
`export CHROME_BIN=/mnt/c/Program\ Files\ \(x86\)/Google/Chrome/Application/chrome.exe`

Alternatively, you could also install a Linux headless chrome by globally installing puppeteer `npm i -g puppeteer` and its [necessary dependencies](https://github.com/puppeteer/puppeteer/issues/3443#issuecomment-433096772) and setting `CHROME_BIN` to that instead.

Both methods worked for me, although the first is definitely the simpler approach.

**Reply by ghost** [reply]
It's been a couple of months since I've run on windows, it would be a good time to kick the tires. I don't use `dev` build either, I just rebuild from inside the model-viewer package to limit the scope of the rebuild.
